src/oca_pre_commit_hooks/checks_odoo_module_xml.py

```diff
@@ -348,3 +348,18 @@ def check_xml_xpath(self):
                         f'{manifest_data["filename_short"]}:{xpath_node.sourceline} '
                         f"Use of translatable xpath `text()`"
                     )
+
+    @utils.only_required_for_checks("xml-oe-structure-missing-id")
+    def check_xml_oe_structure(self):
+        """* Check xml-oe-structure-missing-id
+
+        Ensure all tags with class 'oe_structure' have an ID. For more information on the rationale see:
+        https://github.com/OCA/odoo-pre-commit-hooks/issues/27
+        """
+        for manifest_data in self.manifest_datas:
+            for xpath_node in manifest_data["node"].xpath("//*[contains(@class, 'oe_structure')]"):
+                if not xpath_node.get("id"):
```

Comment:
This could be considered in the xpath, e.g.

`//*[hasclass('oe_structure') and not(@id)]`

That way, you won't need this condition